lib/core-integration/src/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs

```diff
@@ -1802,6 +1806,24 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
                         withdrawal <- PlutusScenario.withdrawScript_1
                         pure (withdrawal, [])
                   )
+                , ( "currency", \ctx w -> do
+                    (addr,proxy) <- view #id . head <$> listAddresses @n ctx w
+                    let getFreshUTxO = do
+                            -- To obtain a fresh UTxO, we perform
+                            -- coin selection and just pick the first input
+                            -- that has been selected.
+                            let singleton = pure -- specialized to NonEmpty
+                            (_, result) <- selectCoins @_ @'Shelley ctx w $
+                                singleton $ AddressAmount
+                                    { address = (addr, proxy)
+                                    , amount  = Quantity 10_000_000
+                                    , assets  = ApiT TokenMap.empty
+                                    }
+                            pure $ head . view #inputs <$> result
+                    txOutRef <- fromEither =<< getFreshUTxO
+                    let mint = PlutusScenario.currencyTx txOutRef
+                    pure (mint, [])
```

Comment:
Curretly all Plutus tx tests are wired to the same "Plutus scenarios" sub-suite which is neat but doesn't give opportunity for more sophisticated assertions. Like, for _Currency_ test, we could check that "apfel" and "banana" are really minted and on wallet balance after the transaction. What do you think?

I am trying to add more assertions though when adjusting contract for e2e tests (e.g. [here](https://github.com/input-output-hk/cardano-wallet/blob/master/test/e2e/spec/e2e_spec.rb#L122-L159)) so maybe that's enough.

Reply:
> Like, for Currency test, we could check that "apfel" and "banana" are really minted and on wallet balance after the transaction. What do you think?

I was wondering the same thing. To be honest, I think that your e2e tests look a lot cleaner than the Haskell version. 😅 I'm not sure exactly why, but I believe the reason is that the Haskell version has a hard time constructing and deconstructing JSON values.

If you ask me, I would rather spend the time on improving the integration test DSL as a whole rather than trying to improve this individual test. In this case, the smart contract checks that the assets are minted; thus, as long as the smart contract is touched by the node, we can be certain that these values are minted, as the node would reject the transaction otherwise.